lib/plug/parsers.ex

```diff
@@ -184,8 +184,10 @@ defmodule Plug.Parsers do
 
   defp reduce(conn, [h|t], type, subtype, headers, opts) do
     case h.parse(conn, type, subtype, headers, opts) do
-      {:ok, body, %Conn{query_params: query} = conn} ->
+      {:ok, body, %Conn{params: %Plug.Conn.Unfetched{}, query_params: query} = conn} ->
         %{conn | body_params: body, params: Map.merge(query, body)}
+      {:ok, body, %Conn{params: params, query_params: query} = conn} ->
+        %{conn | body_params: body, params: Enum.reduce([body, query, params], &Map.merge/2)}
```

Comment:
I believe it should be `[query, body, params]` as the body wins over query in the line above too.